docs/requirements/software-requirements.md

```diff
@@ -0,0 +1,234 @@
+(including AWS, Azure, Google and DigitalOcean) as well as on a developer’s
+local Mac or Windows machine.
+
+(This corresponds to [point 1](infrastructure-requirements.md#1-the-infrastructure-runs-applications-that-are-packaged-as-oci-containers-must-have-sep) of the [Infrastructure Requirements (IR)](infrastructure-requirements.md))
+
+1. The container images *MUST* be published in CLARIAH's container registry at each release. This can be automated (see
+   [IR7](infrastructure-requirements.md#7-the-infrastructure-automatically-builds-the-application-must-have-auto)).
+2. Containers *SHOULD* be kept as lightweight application containers, not including unnecessary runtime components.
+3. Configuration values (such as database connection strings or API URLs) that may vary between deployments
+   *SHOULD* be parameters to the container. This is implemented through environment variables. The infrastructure in
+   turn configures applications through environment variables (see [IR4](infrastructure-requirements.md#4-the-infrastructure-configures-applications-through-environment-variables-must-have-sep) and also related to point 7).
+4. Containers should output all log information to ``stdout`` so it can be captured by the infrastructure (see IR5 and
+   IR6).
+5. Application data (state) that needs to be persistant between runs *MUST* be stored separate from the container (e.g. in a
+   mounted volume) (See point 3 of the [Infrastructure Requirements (IR)](infrastructure-requirements.md))
+
+### 13. Service developers *SHOULD* provide an initial template when of multi-container orchestration is needed
+
+If a complex service consists of multiple interacting containers, the developers *SHOULD* provide an initial template in
+the form of a Docker Compose configuration or a Kubernetes deployment configuration that illustrates how the containers
+are orchestrated to form the application. This *MUST* be maintained in a VCS repository (infrastructure as code
+principle). Infrastructure operators can build on this example to deploy the application.
+
+### 14. Services *MUST* be compatible with CLARIAH's authentication and authorization infrastructure
+
+All services open to end-users and which require some form of user authentication should be compatible with
+CLARIAH's authentication and authorization infrastructure. That is, they should be able to communicate with CLARIAH's
+[SATOSA](https://github.com/IdentityPython/SATOSA) Authentication Provider. It is *RECOMMENDED* to use OpenID Connect
+for this communication. Instruction can be found [(TODO add link!)]().
+
+### 15. Services *MUST* expose a public endpoint providing their specification
+
+1. Swagger, WADL, and CLAM are *SUGGESTED* as possible interface description languages.
+2. The endpoint *SHOULD NOT* be hindered by any authentication barriers.
+
+### 16. Services *SHOULD* expose a public endpoint providing high-level codemeta metadata
```

Comment:
```suggestion
### 16. Services *SHOULD* expose a public endpoint providing high-level CodeMeta metadata
```

Is any standard URL for this endpoint described by CodeMeta (cf. the `/metrics` endpoint for Prometheus/OpenTelemetry)?

Reply:
No, CodeMeta doesn't define any of that, we'll have to make a choice ourselves.